include/swift/Runtime/Metadata.h

```diff
@@ -2623,7 +2623,8 @@ SWIFT_RUNTIME_EXPORT
 ClassMetadata *
 swift_allocateGenericClassMetadata(GenericMetadata *pattern,
                                    const void *arguments,
-                                   ClassMetadata *superclass);
+                                   ClassMetadata *superclass,
+                                   size_t numImmediateMembers);
```

Comment:
I think the original idea here was that class-specific data that's constant for all instantiations should be derived from the pattern.  But that might be awkward if it means introducing subclasses of GenericMetadata, since the template currently immediately follows the header.  On the other hand, I'm not thrilled with the design of GenericMetadata as something we want to ABI-commit to, so if it's convenient...

Reply:
Yeah, adding another field to the template header seemed awkward to do because of alignment.